### PR TITLE
Make SonarTS independent of workdir location

### LINKED ISSUE #195: Make SonarTS independent of workdir location
An issue with TFS ( #190 ) shows that using the workdir to deploy sonarts-core is fragile because we rely on workdir being within the target project to allow for typescript compiler resolution.

On possible solution is to deploy sonarts-core into the sources folder, but it needs to be thought out.

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #195

### CODE REVIEW

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java [hunk 1]
```diff
@@ -110,11 +114,11 @@ public void execute(SensorContext sensorContext) {
     LOG.info("Rules execution");
     TypeScriptRules typeScriptRules = new TypeScriptRules(checkFactory);
     executableBundle.activateRules(typeScriptRules);
-    runRules(inputFiles, executableBundle, sensorContext, typeScriptRules, deployDestination);
+    runRules(inputFiles, executableBundle, sensorContext, typeScriptRules, deployDestination, typescriptLocation);
 
   }
 
-  private void runRules(Iterable<InputFile> inputFiles, ExecutableBundle executableBundle, SensorContext sensorContext, TypeScriptRules typeScriptRules, File deployDestination) {
+  private void runRules(Iterable<InputFile> inputFiles, ExecutableBundle executableBundle, SensorContext sensorContext, TypeScriptRules typeScriptRules, File deployDestination, @Nullable File typescriptLocation) {
```
Comment (SonarTech) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java]: MINOR Split this 213 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java [hunk 2]
```diff
@@ -328,6 +339,37 @@ private void saveHighlights(SensorContext sensorContext, Highlight[] highlights,
     highlighting.save();
   }
 
+
+  @Nullable
+  private static File getTypescriptLocation(File topDir) {
+    File node_modules = getDir(topDir, "node_modules");
```
Comment (SonarTech) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java]: MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00117)

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java [hunk 3]
```diff
@@ -27,6 +27,7 @@
 import com.google.gson.Gson;
 import java.io.File;
 import java.io.FileInputStream;
+import java.io.FilenameFilter;
```
Comment (SonarTech) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java]: MINOR Remove this unused import 'java.io.FilenameFilter'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java [hunk 4]
```diff
@@ -328,6 +343,37 @@ private void saveHighlights(SensorContext sensorContext, Highlight[] highlights,
     highlighting.save();
   }
 
+
+  @Nullable
+  private static File getTypescriptLocation(File topDir) {
```
Comment (inverno) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java]: @vilchik-elena topDir -> currentDir

Comment (inverno) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java]: @vilchik-elena what about extracting this in its own class and writing a dedicated unit test?

**Pull request merged**